src/vote_signer_proxy.rs

```diff
@@ -26,27 +28,65 @@ pub enum VoteError {
     LeaderInfoNotFound,
 }
 
-pub struct VoteSignerProxy {
+pub struct VoteSignerForwarder {
     rpc_client: RpcClient,
+}
+
+impl VoteSignerForwarder {
+    pub fn new(signer: SocketAddr) -> Self {
+        Self {
+            rpc_client: RpcClient::new_from_socket(signer),
+        }
+    }
+}
+
+impl VoteSignerInterface for VoteSignerForwarder {
+    fn register(
+        &self,
+        pubkey: Pubkey,
+        sig: &Signature,
+        msg: &[u8],
+    ) -> jsonrpc_core::Result<Pubkey> {
+        let params = json!([pubkey, sig, msg]);
+        let resp = RpcRequest::RegisterNode
+            .retry_make_rpc_request(&self.rpc_client, 1, Some(params), 5)
+            .unwrap();
+        let vote_account: Pubkey = serde_json::from_value(resp).unwrap();
+        Ok(vote_account)
+    }
+    fn sign(&self, pubkey: Pubkey, sig: &Signature, msg: &[u8]) -> jsonrpc_core::Result<Signature> {
+        let params = json!([pubkey, sig, msg]);
+        let resp = RpcRequest::SignVote
+            .make_rpc_request(&self.rpc_client, 1, Some(params))
+            .unwrap();
+        let vote_signature: Signature = serde_json::from_value(resp).unwrap();
+        Ok(vote_signature)
+    }
+    fn deregister(&self, pubkey: Pubkey, sig: &Signature, msg: &[u8]) -> jsonrpc_core::Result<()> {
+        let params = json!([pubkey, sig, msg]);
+        let _resp = RpcRequest::DeregisterNode
+            .retry_make_rpc_request(&self.rpc_client, 1, Some(params), 5)
+            .unwrap();
+        Ok(())
+    }
+}
+
+pub struct VoteSignerProxy {
     keypair: Arc<Keypair>,
+    signer: Box<VoteSignerInterface + Send + Sync>,
```

Comment:
Doesn't look like dynamic dispatch should be required here.  Have you tried removing the `Box` and making that `T` where `T: VoteSignerInterface + Send + Sync`?

Reply:
If you find that you can remove the `Arc` in the fullnode constructor, and that you can *move* this object straight to the one location its needed, then you should be able to remove that `Sync` constraint too.

Reply:
I'll try to remove Box, and use T.
I had to add Send/Sync because the object is being passed from fullnode main thread to replay_stage thread. Do you think we can do away with Sync?

Reply:
```VoteSignerProxy``` is storing the dynamic reference in it's structure. Can this be done via a static dispatch as well? I could not find an example of it.

```
pub struct VoteSignerProxy {
    keypair: Arc<Keypair>,
    signer: Box<VoteSigner + Send + Sync>,
    pub vote_account: Pubkey,
}
```

Reply:
I might be wrong, but I think we do need dynamic dispatch.

The trait object for VoteSigner is being passed to VoteSignerProxy, that stores it in proxy object. The proxy object is being passed to Fullnode::new() and few other functions. The binding of the trait object is dynamic (i.e. for test we are passing LocalVoteSigner trait object, and for non-test we are passing RemoteVoteSigner trait object).

If we were to use static dispatch, we would need to change all these downstream objects to traits as well. Is that right?